Update show jobs doc for changefeed retention time (#17044)
cockroachdb/docs: _includes/v23.1/cdc/show-changefeed-job-retention.md

> {% include_cached new-in.html version="v23.1" %} All changefeed jobs will display regardless of if the job completed and when it completed. You can use the [cluster setting](cluster-settings.html) `jobs.retention_time` to define a retention time to delete completed jobs.

Reviewer: ```suggestion
{% include_cached new-in.html version="v23.1" %} All changefeed jobs will display regardless of if the job completed and when it completed. You can define a retention time and delete completed jobs by using the `jobs.retention_time` [cluster setting](cluster-settings.html).
```
